packages/core/useFileDialog/index.ts

```diff
@@ -90,47 +90,82 @@ export function useFileDialog(options: UseFileDialogOptions = {}): UseFileDialog
   const files = deepRef<FileList | null>(prepareInitialFiles(options.initialFiles))
   const { on: onChange, trigger: changeTrigger } = createEventHook()
   const { on: onCancel, trigger: cancelTrigger } = createEventHook()
-  let input: HTMLInputElement | undefined
-  if (document) {
-    input = unrefElement(options.input) || document.createElement('input')
-    input.type = 'file'
-
-    input.onchange = (event: Event) => {
-      const result = event.target as HTMLInputElement
-      files.value = result.files
-      changeTrigger(files.value)
+  const inputRef = computed(() => {
+    const input = unrefElement(options.input) ?? (document ? document.createElement('input') : undefined)
+    if (input) {
+      input.type = 'file'
+
+      input.onchange = (event: Event) => {
+        const result = event.target as HTMLInputElement
+        files.value = result.files
+        changeTrigger(files.value)
+      }
+
+      input.oncancel = () => {
+        cancelTrigger()
+      }
     }
-
-    input.oncancel = () => {
-      cancelTrigger()
-    }
-  }
+    return input
+  })
 
   const reset = () => {
     files.value = null
-    if (input && input.value) {
-      input.value = ''
+    if (inputRef.value && inputRef.value.value) {
+      inputRef.value.value = ''
       changeTrigger(null)
     }
   }
 
+  /**
+   * Apply composable state to the element, also when element is changed
+   */
+  watch([inputRef, () => toValue(options.multiple)], () => {
+    const el = inputRef.value
+    if (!el)
+      return
+    el.multiple = toValue(options.multiple)!
+  }, { immediate: true })
+
+  watch([inputRef, () => toValue(options.accept)], () => {
+    const el = inputRef.value
+    if (!el)
+      return
+    el.accept = toValue(options.accept)!
+  }, { immediate: true })
+
+  watch([inputRef, () => toValue(options.directory)], () => {
+    const el = inputRef.value
+    if (!el)
+      return
+    // webkitdirectory key is not stabled, maybe replaced in the future.
+    el.webkitdirectory = toValue(options.directory)!
+  }, { immediate: true })
+
+  watch([inputRef, () => toValue(options.capture)], () => {
+    const el = inputRef.value
+    if (!el)
+      return
+    el.capture = toValue(options.capture)!
+  }, { immediate: true })
```

Comment:
```suggestion
  watch([
    inputRef,
    () => toValue(options.multiple),
    () => toValue(options.accept),
    () => toValue(options.directory),
    () => toValue(options.capture),
  ], () => {
  	// update
  }, { immediate: true })
```

Can we use one `watch` to handle this? Then add an update function to update these properties.